frontend/dfx_react_typescript/src/dfx_react_typescript/public/App.tsx

```diff
@@ -0,0 +1,58 @@
+import reactTSCanister from 'ic:canisters/dfx_react_typescript';
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+interface AppState {
+  count: number;
+}
+interface AppProps {}
+
+class App extends React.Component<AppProps, AppState> {
+  public state: AppState = {
+    count: 0,
+  };
+
+  constructor(props: AppProps) {
+    super(props);
+  }
+
+  public increment = this._updateCount.bind(this, '+');
```

Comment:
Why not just have

```typescript
public async increment() {
  this.setState({ count: +(await reactTSCanister.increment()) });
}
```

The binding and the switch/case below is a lot of code.

Reply:
Either way you have to bind the event to the `Component` class, but I can refactor to try and make this simpler